### PR TITLE
WT-8503 Create a function to execute csuite tests in evergreen.yml

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -413,14 +421,6 @@ functions:
...
         ${test_env_vars|} ./t ${thread_test_args|} 2>&1
-  "random abort test":
-    command: shell.exec
-    params:
-      working_dir: "wiredtiger/build_posix/test/csuite"
-      script: |
-        set -o errexit
-        set -o verbose
-        ${test_env_vars|} ./test_random_abort ${random_abort_args|} 2>&1
   "tiered abort test":
```
Comment (lukech) [inline test/evergreen.yml:424]: Can the following abort test functions (tiered, timstamp, etc) be deprecated in favor of the new `csuite test` function?

File: test/evergreen.yml [hunk 2]
```diff
@@ -358,6 +358,14 @@ functions:
...
           ${test_env_vars|} ${make_command|make} VERBOSE=1 check ${smp_command|} 2>&1
         fi
+  "csuite test":
+    command: shell.exec
+    params:
+      working_dir: "wiredtiger/build_posix/test/csuite"
+      script: |
+       set -o errexit
+       set -o verbose
+       ${test_env_vars|} ./${test_exe} ${test_args|} 2>&1
```
Comment (lukech) [inline test/evergreen.yml:368]: A nit for the naming of `test_exe`, which makes me think of a Windows-specific executable? Can it be renamed to e.g. "test_app" or "test_exec" instead?

**Review by lukech — COMMENTED** [review]
This PR saves so many LOC in our evergreen yaml file. Good job!

Hope to point out that we have `dist/s_evergreen` to remind users to add a new Evergreen task once a new test is added into the `test/csuite` directory, and the template file `test/evergreen/csuite_test_evg_task.template` may need to be updated to the new style by using `csuite test` function.

File: test/evergreen.yml (hunk 1 above)
Comment (etienneptl) [reply test/evergreen.yml:424]: Some of them are actually unused and we are deciding what to do with them as part of WT-8504.

**Comment by etienneptl** [comment]
> This PR saves so many LOC in our evergreen yaml file. Good job!
> 
> Hope to point out that we have `dist/s_evergreen` to remind users to add a new Evergreen task once a new test is added into the `test/csuite` directory, and the template file `test/evergreen/csuite_test_evg_task.template` may need to be updated to the new style by using `csuite test` function.

@lukech, thanks for pointing out those files, I was not aware of them. I update the template in 94da98c585e4330022273061c01925dd00829ab3.

File: test/evergreen.yml (hunk 2 above)
Comment (etienneptl) [reply test/evergreen.yml]: I renamed it to `test_binary` as I saw Evergreen was using the following syntax for binaries `xxx_binary`. See the changes in cb9823ae1b605e68ddb7ad62ac4aa98e71db6c76.

**Review by Siddhartha8899 — APPROVED** [review]

**Review by lukech — APPROVED** [review]

**Pull request merged**